projects/buck2/build.mk

```diff
@@ -0,0 +1,50 @@
+# --------------------------------------------------------------------------------
+# Project: Buck2 Build Script
+# Description: This script is used for building and managing the Buck2 project.
+# Author: Christian Abrokwa
+# --------------------------------------------------------------------------------
+
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+
+$(eval $(call project-define,buck2))
```

Comment:
I am not sure whether `project-define` helps at all do what we want here, we don't want sources, we don't need build dir, perhaps it would be a good idea to write make rules that avoid this helper call. How about structuring this Makefile as follows:

```
BUCK2_VERSION = ...
BUCK2_ARCHIVE = ...
BUCK2_URL = ...

buck2-host: $(HOST_OUT_DIR)/bin/buck2

$(HOST_OUT_DIR)/bin/buck2: $(DOWNLOADS_DIR)/$(BUCK2_ARCHIVE) | $(HOST_OUT_DIR)
    # commands to unpack $(BUCK2_ARCHIVE) and set the executable flag

$(DOWNLOADS_DIR)/$(BUCK2_ARCHIVE) | $(DOWNLOADS_DIR)
    # instructions to download the archive

```

I suggest making the downloaded file an [order-only prerequisite](https://www.gnu.org/software/make/manual/html_node/Prerequisite-Types.html) (that is what | is for).